### ISSUE TITLE
Regression: Fragment changes do not handle focus correctly

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/firefox-tv#1888]
## Steps to reproduce
Open Pocket or Settings fragment

### Expected behavior
Item is focused

### Actual behavior
Nothing is focused

### Device information
* Fire TV device: Stick
* Latest affected Firefox version: master e5a56a9f4e67e253d6e3786d25d9291dd3958f41
* Earliest affected Firefox version: 3.4.1 master

I think this is a regression from #1664 due to changing focus between webrender/overlay fragment, but not handling other fragments.

**Comment by liuche** [comment]
This needs a focus viewmodel that depends on the `currentActiveScreen` livedata. I've got this mostly working, except for getting YT focus.

**Comment by liuche** [comment]
This fix ended up being a problem with a) which fragment to dispatch keypresses to, and b) focus change on Fragment change.

Focus:
At first, I tried observing the `currentActiveScreen` to decide which Fragment to focus, but that ViewModel state actually updates before the fragment is shown - we did this in #1830 because we needed to cache the WebView focus right before active screen change in order to get the correct element. Additionally, focus change isn't strictly a ViewModel change, and depends on the Fragment's view visibility (which is handled asynchronously by the FragmentManager). I ended up adding the focus request in the Fragment callbacks themselves, in `onHiddenChanged`.

Keypress Dispatch:
Additionally, we were not doing correct handling of keypresses, and sometimes sending them to the wrong Fragment. Our key handling is: handle special keys (by triggering fragment changes or sending them to specific fragments), or allow them to fall through to the system, which handle things like navigating around Android UI elements (in a RecyclerView for example).
